source/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter.cc

```diff
@@ -299,34 +365,39 @@ Http::FilterHeadersStatus JsonTranscoderFilter::decodeHeaders(Http::RequestHeade
     // just pass-through the request to upstream.
     return Http::FilterHeadersStatus::Continue;
   }
-  has_http_body_output_ = !method_->server_streaming() && hasHttpBodyAsOutputType();
+  has_http_body_output_ =
+      !method_->descriptor_->server_streaming() && method_->output_type_is_http_body_;
+  if (method_->input_type_is_http_body_) {
+    if (headers.ContentType() != nullptr) {
+      absl::string_view content_type = headers.ContentType()->value().getStringView();
+      content_type_.assign(content_type.begin(), content_type.end());
+    }
+
+    bool done = !readToBuffer(*transcoder_->RequestOutput(), request_prefix_);
+    RELEASE_ASSERT(done, "request transcoder should have been done");
```

Comment:
Just reject the request with error log? I don't think RELEASE_ASSERT should be done in data path of a filter.